### DISCUSSION TITLE
Even name to be used when setup a  add_callback for mouse pick event

### DISCUSSION THREAD

**Discussion by DRLing2021** [marcomusy/vedo#750] [Q&A] [answered]
We are proposing to monitor the mouse click event to show the coordinates of picked pt from 3D point cloud. The plan is to use add_callback function from Plotter Class. From the document, the method definition is :

def add_callback (self, eventName, func, priority=0.0)

Furthermore, there is an example script shown in the document, as : 
```python
from vedo import *

def func(evt): # called every time the mouse moves
    # evt is a dotted dictionary
    if not evt.actor:
        return  # no hit, return
    print("point coords =", evt.picked3d)
    # print("full event dump:", evt)

elli = Ellipsoid()
plt = Plotter(axes=1)
plt.add_callback('mouse hovering', func)
plt.show(elli).close()
```
It seems that the evenName is "mouse hover" in the example.  However the frequently used event names listed in the same document are: 

- KeyPress, KeyRelease: listen to keyboard events
- LeftButtonPress, LeftButtonRelease: listen to mouse clicks
- MiddleButtonPress, MiddleButtonRelease
- RightButtonPress, RightButtonRelease
- MouseMove: listen to mouse pointer changing position
- MouseWheelForward, MouseWheelBackward
- Enter, Leave: listen to mouse entering or leaving the window
- Pick, StartPick, EndPick: listen to object picking
- ResetCamera, ResetCameraClippingRange
- Error, Warning
- Char
- Timer

No “mouse hover” in the list Are there anything wrong? As for our case that is to show the “mouse picked” pt's coordinates, what is the right "eventName" to be used when setup add_callback? 

Thanks in advance!
Regards,
DR, Ling

**Comment by ZiguoAtGitHub** [comment, marked as answer]
> We are proposing to monitor the mouse click event to show the coordinates of picked pt from 3D point cloud. The plan is to use add_callback function from Plotter Class. From the document, the method definition is :
> 
> def add_callback (self, eventName, func, priority=0.0)
> 
> Furthermore, there is an example script shown in the document, as :
> 
> from vedo import *
> 
> def func(evt): # called every time the mouse moves # evt is a dotted dictionary if not evt.actor: return # no hit, return print("point coords =", evt.picked3d) # print("full event dump:", evt)
> 
> elli = Ellipsoid() plt = Plotter(axes=1) plt.add_callback('mouse hovering', func) plt.show(elli).close()
> 
> It seems that the evenName is "mouse hover" in the example. However the frequently used event names listed in the same document are:
> 
> * KeyPress, KeyRelease: listen to keyboard events
> * LeftButtonPress, LeftButtonRelease: listen to mouse clicks
> * MiddleButtonPress, MiddleButtonRelease
> * RightButtonPress, RightButtonRelease
> * MouseMove: listen to mouse pointer changing position
> * MouseWheelForward, MouseWheelBackward
> * Enter, Leave: listen to mouse entering or leaving the window
> * Pick, StartPick, EndPick: listen to object picking
> * ResetCamera, ResetCameraClippingRange
> * Error, Warning
> * Char
> * Timer
> 
> No “mouse hover” in the list Are there anything wrong? As for our case that is to show the “mouse picked” pt's coordinates, what is the right "eventName" to be used when setup add_callback?
> 
> Thanks in advance! Regards, DR, Ling

LeftButtonRelease Event?

**Reply by marcomusy** [reply]
Hi, yes there are a number of automatic aliases because I find them hard to remember...
So for ex: "LeftButtonPress" or "left mouse click" or "please call my function when hitting left button" are all equivalent and valid :)
